pkg/fqdn/matchpattern/matchpattern_test.go

```diff
@@ -58,6 +64,8 @@ func (ts *MatchPatternTestSuite) TestUnAnchoredMatchPatternREConversion(c *C) {
 // *.cilium.io. matches anysub.cilium.io. but not cilium.io.
 // *cilium.io. matches  anysub.cilium.io. and cilium.io.
 // *.ci*.io. matches anysub.cilium.io. anysub.ci.io., anysub.ciliumandmore.io. but not cilium.io.
+// **.cilium.io. matches  anysub.cilium.io. andmomre subdomains.
+// **.ci*.io. matches anysub.cilium.io. anysub.ci.io. and more subdomains anysub.ciliumandmore.io. but not cilium.io.
```

Comment:
It seems relevant to call out that it also matches `some.more.ci.io`, or did I missunderstand the point of the PR?